File: Age Distribution.m
#import <Foundation/Foundation.h>


NSString* getDistribution(NSString *line) {
    int tmpInt = [line intValue];
    
    if(tmpInt < 0 || tmpInt > 100)
        return @"This program is for humans";
    else if(tmpInt >= 66)
        return @"Retirement";
    else if(tmpInt >= 23)
        return @"Work";
    else if(tmpInt >= 19)
        return @"College";
    else if(tmpInt >= 15)
        return @"High school";
    else if(tmpInt >= 12)
        return @"Middle school";
    else if(tmpInt >= 5)
        return @"Elementary school";
    else if(tmpInt >= 3)
        return @"Preschool";
    else if(tmpInt >= 0)
        return @"Home";
    
    return nil;
}


int main(int argc, const char * argv[])
{

    @autoreleasepool {
        NSString *filename = [NSString stringWithCString:argv[1] encoding:NSASCIIStringEncoding];
        NSString *content = [NSString stringWithContentsOfFile:filename encoding:NSASCIIStringEncoding error:nil];
        NSArray *lines = [content componentsSeparatedByString:@"\n"];
        
        for(NSString *line in lines) {
            NSString *tmp = getDistribution(line);
            NSLog(@"%@", tmp);
        }
    }
    return 0;
}

File: Age Distribution.rb
def calculate(line)
  int = line.to_i
  
  if(int < 0 || int > 100)
    "This program is for humans"
  elsif(int >= 66)
    "Retirement"
  elsif(int >= 23)
    "Work"
  elsif(int >= 19)
    "College"
  elsif(int >= 15)
    "High school"   
  elsif(int >= 12)
    "Middle school"
  elsif(int >= 5)
    "Elementary school"
  elsif(int >= 3)
    "Preschool"
  elsif(int >= 0)
    "Home"
  end 
end



File.open(ARGV[0]).each do |line|
  puts calculate(line)
end